scripts/zephyr_module.py

```diff
@@ -176,37 +176,24 @@ def main():
                              list`""")
     parser.add_argument('-x', '--extra-modules', nargs='+', default=[],
                         help='List of extra modules to parse')
-    parser.add_argument('-w', '--west-path', default='west',
-                        help='Path to west executable')
     parser.add_argument('-z', '--zephyr-base',
                         help='Path to zephyr repository')
     args = parser.parse_args()
 
     if args.modules is None:
-        p = subprocess.Popen([args.west_path, 'list', '--format={posixpath}'],
-                             cwd=args.zephyr_base,
-                             stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE)
-        out, err = p.communicate()
-        if p.returncode == 0:
-            projects = out.decode(sys.getdefaultencoding()).splitlines()
-        elif re.match((r'Error: .* is not in a west installation\.'
-                       '|FATAL ERROR: no west installation found from .*'
-                       '|FATAL ERROR: no west workspace.*'),
-                      err.decode(sys.getdefaultencoding())):
-            # Only accept the error the event we are outside a west
-            # workspace.
-            #
-            # TODO: we can just use "west topdir" instead if we can
-            # depend on west 0.7.0 or later.
-            projects = []
-        else:
-            print(err.decode(sys.getdefaultencoding()))
-            # A real error occurred, raise an exception
-            raise subprocess.CalledProcessError(cmd=p.args,
-                                                returncode=p.returncode)
-    else:
-        projects = args.modules
+        # West is imported here, as it is optional (and thus maybe not installed)
+        # if user is providing a specific modules list.
+        from west.manifest import Manifest
+        from west.util import WestNotFound
+        try:
+            manifest = Manifest.from_file()
+            projects = [p.posixpath for p in manifest.get_projects([])]
+            print(projects)
```

Comment:
Shouldn't this be `print('\n'.join(projects))` or so?

Reply:
Wait, actually, the previous version doesn't seem to print the projects list. Is this just leftover debugging code?

Reply:
Leftover debugging code removed.